lua/mason-core/managers/git/init.lua

```diff
@@ -27,8 +27,7 @@ function M.clone(opts)
         ".",
     }
     _.coalesce(opts.version, ctx.requested_version):if_present(function(version)
-        ctx.spawn.git { "fetch", "--depth", "1", "origin", version }
-        ctx.spawn.git { "checkout", "FETCH_HEAD" }
+        ctx.spawn.git { "switch", "-c", version }
```

Comment:
> Hm this interaction with git should not have been the issue. `git switch` is somewhat new (although time is relative) and only exists in newer versions of git, it’s also considered unstable. Let’s revert this to `fetch -> checkout` flow again.
> 
> What leads you to believe this has been an issue?

Because switch works, and checkout fetchhead failed...

Reply:
> Failed how?

![image](https://user-images.githubusercontent.com/60290287/212330303-55a2eb0a-15d7-4926-83ee-3ff813a9ad30.png)

Above , I use switch, under I use git fetch, and above, it passed

Emm. I see, I have tried the release, it also failed, I am using master branch, I am sorry

Reply:
> Failed how?

Then I will tell autor to make a new tag.. now, git version will work